cgroups/fs/apply_raw.go

```diff
@@ -128,14 +128,22 @@ func (m *Manager) Apply(pid int) error {
 }
 
 func (m *Manager) Destroy() error {
-	return cgroups.RemovePaths(m.Paths)
+	m.mu.Lock()
+	paths := m.Paths
+	m.mu.Unlock()
+	return cgroups.RemovePaths(paths)
```

Comment:
i actually prefer to lock for entire duration of Destroy and reset Paths in the end
 
```go
m.mu.Lock()
defer m.mu.Unlock()
err := cgroups.RemovePaths(paths)
if err != nil {
  return err
}
m.Paths = make(map[string]string)
return nil
```

So the race between Destroy and GetStats will be removed. `Stats` calls in the middle of `Destroy` will wait and not return partial data. Wdyt ?